sdk/tests/js/tests/tex-image-and-sub-image-utils.js

```diff
@@ -791,11 +791,71 @@ var TexImageUtils = (function() {
     return program;
   };
 
+  /**
+   * Return a list of unpack color spaces to test, supported by the specified
+   * WebGLRenderingContext.
+   */
+  var unpackColorSpacesToTest = function(gl)
+  {
+    if ('unpackColorSpace' in gl)
+      return ['srgb', 'display-p3'];
+    else
+      return [null];
+  }
+
+  /**
+   * For each entry in unpackColorSpaces, duplicate all of cases, adding an
+   * unpackColorSpace key with its value set to that entry to each case.
+   */
+  var crossProductTestCasesWithUnpackColorSpaces = function(cases, unpackColorSpaces)
+  {
+    var caseWithColorSpace = function(c, cs) { return Object.assign({}, c, {unpackColorSpace:cs}); }
+    var casesList = unpackColorSpaces.map(cs => cases.map(c => caseWithColorSpace(c, cs)));
+    return [].concat.apply([], casesList);
```

Comment:
Could this just be

```
return caseList;
```

?

Reply:
casesList ends up being
  [[list of cases with srgb], [list of cases with display-p3]]
and so the concat moves them all into a single list.

Reply:
In that case you can use `flat`

```
return caseList.flat()
```

Reply:
Updated.